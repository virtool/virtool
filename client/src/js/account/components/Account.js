import React, { useEffect } from "react";
import { connect } from "react-redux";
import { Switch, Redirect, Route } from "react-router-dom";

import { NarrowContainer, TabLink, Tabs, ViewHeader, ViewHeaderTitle, WideContainer } from "../../base";
import { getAccount } from "../actions";
import { getAccountId } from "../selectors";
import AccountProfile from "./Profile";
import APIKeys from "./API/API";

export const Account = ({ userId, onGet }) => {
    useEffect(() => onGet(), [userId]);

    return (
        <WideContainer>
            <ViewHeader title="Account">
                <ViewHeaderTitle>Account</ViewHeaderTitle>
            </ViewHeader>

            <Tabs>
                <TabLink to="/account/profile">Profile</TabLink>
                <TabLink to="/account/api">API</TabLink>
            </Tabs>

            <NarrowContainer>
                <Switch>
                    <Redirect from="/account" to="/account/profile" exact />
                    <Route path="/account/profile" component={AccountProfile} />
                    <Route path="/account/api" component={APIKeys} />
                </Switch>
            </NarrowContainer>
        </WideContainer>
    );
};

export const mapStateToProps = state => ({
    userId: getAccountId(state)
});

export const mapDispatchToProps = dispatch => ({
    onGet: () => {
        dispatch(getAccount());
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Account);
