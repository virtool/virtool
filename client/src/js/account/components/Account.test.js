import Account from "./Account";

describe("<Account />", () => {

    it("renders correctly", () => {
        const initialState = {
            account: {}
        };
        const store = mockStore(initialState);

        const container = shallow(<Account store={store} />).dive();

        expect(container).toMatchSnapshot();
    });
});
