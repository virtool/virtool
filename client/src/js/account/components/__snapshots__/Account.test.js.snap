// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> renders correctly 1`] = `
<div
  className="container"
>
  <h3
    className="view-header"
  >
    <strong>
      Account
    </strong>
  </h3>
  <Nav
    bsClass="nav"
    bsStyle="tabs"
    justified={false}
    pullLeft={false}
    pullRight={false}
    stacked={false}
  >
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/account/general"
    >
      <NavItem
        active={false}
        disabled={false}
      >
        General
      </NavItem>
    </LinkContainer>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/account/settings"
    >
      <NavItem
        active={false}
        disabled={false}
      >
        Settings
      </NavItem>
    </LinkContainer>
    <LinkContainer
      activeClassName="active"
      exact={false}
      replace={false}
      strict={false}
      to="/account/api"
    >
      <NavItem
        active={false}
        disabled={false}
      >
        API
      </NavItem>
    </LinkContainer>
  </Nav>
  <Switch>
    <Redirect
      exact={true}
      from="/account"
      push={false}
      to="/account/general"
    />
    <Route
      component={[Function]}
      path="/account/general"
    />
    <Route
      component={[Function]}
      path="/account/settings"
    />
    <Route
      component={[Function]}
      path="/account/api"
    />
  </Switch>
</div>
`;
